library/src/main/java/com/bumptech/glide/request/transition/BitmapContainerTransitionFactory.java

```diff
@@ -0,0 +1,55 @@
+package com.bumptech.glide.request.transition;
+
+import android.content.res.Resources;
+import android.graphics.Bitmap;
+import android.graphics.drawable.BitmapDrawable;
+import android.graphics.drawable.Drawable;
+
+import com.bumptech.glide.load.DataSource;
+
+/**
+ * A {@link TransitionFactory} for complex types that have a {@link android.graphics.Bitmap} inside.
+ * The transitioning bitmap is wrapped in a {@link android.graphics.drawable.BitmapDrawable}.
+ * Most commonly used with {@link DrawableCrossFadeFactory}.
+ *
+ *  @param <R> The type of the composite object that contains the {@link android.graphics.Bitmap} to
+ *            be transitioned.
+ */
+public abstract class BitmapContainerTransitionFactory<R> implements TransitionFactory<R> {
```

Comment:
Do you have another use case in mind for this other than BitmapTransitionFactory below?

If not, it might be simpler to combine the two in a concrete class for now. We could extract something more general if there's a use case?

Reply:
@TWiStErRob any chance you know about a more recent solution for crossfades with PaletteBitmap targets? I had originally tried GenericTransitionOptions with a basic fade animation, but it doesn't work with placeholders, which are removed before the animation starts. This solution looked promising (had to make adjustments since it's four years old) but the animation is completely ignored with the above method, even with a custom duration.

Reply:
I think that may be due to the fact that your placeholder and final image are not the same type, so there is no transition between them. I haven't used Glide 4 enough to know more details, it could be a simple behavior change between 3 and 4. Try to look around the factories and targets to see who is removing the placeholder and try to work backwards from there to find how it could be avoided. You could also try to follow your animation object to see where it gets ignored.

Reply:
Thank you so much! It ended up being a stray line of code in `onResourceReady()` for a custom target. I'm now encountering another issue though, and would appreciate any insight if possible. When the images are leave the frame in a RecyclerView and then enter again while scrolling, the placeholder shows up indefinitely. My first theory was that the TransitionDrawable was somehow reset and the initial layer was shown with no animation start. I also thought it might be an issue with the BitmapPool or recycled images. Does either of those sound more plausible based on the symptoms?

Reply:
I was able to find the issue, so I'll add it here for anyone else who might have a similar problem.

```
        @Override
        protected void setResource(PaletteBitmap resource) {
            view.setImageBitmap(resource.bitmap);
        }
```

I had originally moved this line of code to `onResourceReady()` since it was crashing the app in `setResource()` which fixed the crash. It had also broken the animation but I didn't notice that until recently. I realized this might be the issue since it was in your code snippet above, so I added it back with a null check before trying to set the bitmap. This ended up fixing the issue while keeping the animation. I'm still not sure why it will sometimes be null, but that's a problem for another day.

Reply:
Nice. Btw, `null` answer is easy:
https://github.com/bumptech/glide/blob/a3a3a6573addd83a5be031bbd55d29e7514ca446/library/src/main/java/com/bumptech/glide/request/target/ImageViewTarget.java#L56-L96
I've updated the code above.